include/deal.II/matrix_free/operators.h

```diff
@@ -1542,7 +1545,7 @@ namespace MatrixFreeOperators
               .local_element(edge_constrained_indices[j][i]) =
               edge_constrained_values[j][i].first;
           }
-        for (; c < BlockHelper::subblock(dst, j).local_size(); ++c)
```

Comment:
I am not sure if anyone tried to do it but this would not work with a `TrilinosWrappers::Vector` in the past.

Reply:
Out of curiosity, why not?

Reply:
the matrix free code assumes that local_size refers to the number of locally owned dofs, not the number of locally owned dofs + number of ghosts (which is what our trilinos vector assumes).

Reply:
At this place of the code, the vectors would need to be non-ghosted, so in principle the ranges would need to match. However, the Trilinos vector does not have a `local_element()` access function because for performance reasons we cannot afford translating MPI-global to MPI-local indices, so the code would not work with Trilinos vectors anyway. One could fix it by adding an index translation for those vectors.